Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -365,12 +366,23 @@ def longFileLinter : Linter where run := withSetOptionIn fun stx ↦ do
     -- `candidate` is necessarily bigger than `lastLine` and hence bigger than `defValue`
     let candidate := (lastLine / 100) * 100 + 200
     let candidate := max candidate defValue
-    if linterBound < lastLine then
+    -- in this case, the file is longer than the default and also than what the option says
+    if defValue ≤ linterBound && linterBound < lastLine then
       logWarningAt stx <| .tagged linter.style.longFile.name
         m!"This file is {lastLine} lines long, but the limit is {linterBound}.\n\n\
           You can extend the allowed length of the file using \
           `set_option linter.style.longFile {candidate}`.\n\
           You can completely disable this linter by setting the length limit to `0`."
+    else
+    -- finally, the file exceeds the default value, but not the option: we only allow the value
```

Comment:
You *could* phrase this check differently:
- check early if the default value is not a multiple of 100 (which seems desirable regardless, right)
- then, just compare candidate and linterBound + 200 here
Given the first change, you could then make the error message clearer: the limit is *too high* and should be lowered.